pydantic/dataclasses.py

```diff
@@ -201,12 +201,7 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
 
         original_cls = cls
 
-        config_dict = config
-        if config_dict is None:
-            # if not explicitly provided, read from the type
-            cls_config = getattr(cls, '__pydantic_config__', None)
-            if cls_config is not None:
-                config_dict = cls_config
+        config_dict = config or getattr(cls, '__pydantic_config__', None)
```

Comment:
`config or` is not equivalent to `config is not None`, hence the failing test. We also loose the code comment which might be useful. Considering the really small perf. improvement, maybe it's not worth keeping this PR?

Or maybe we change it as:

```suggestion
        config_dict = config if config is not None else getattr(cls, '__pydantic_config__', None)
```

Without talking about performacne, from a readability perspective, this one liner seems better imo